### PR TITLE
add share export locations requests and models

### PR DESCRIPTION
**Pull request by andypf**
@Carthaca Please take a look

### CODE REVIEW

File: lib/fog/shared_file_system/openstack/requests/get_share_export_location.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            :path    => "shares/#{share_id}/export_locations/​{export_location_id}​"
+          )
+        end
+      end
+
+      class Mock
+        def get_share_export_location(id)
+          response = Excon::Response.new
+          response.status = 200
+          share_export_location = data[:share_export_location]
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack/requests/get_share_export_location.rb]: would you please add some Mock data in `lib/fog/shared_file_system/openstack.rb`?

File: lib/fog/shared_file_system/openstack/requests/list_share_export_locations.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          )
+        end
+      end
+
+      class Mock
+        def list_share_export_locations(share_id)
+          response = Excon::Response.new
+          response.status = 200
+
+          locations = data[:export_locations]
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack/requests/list_share_export_locations.rb]: same as above

**Comment by Carthaca** [comment]
maybe comment or make sure technically (have a look at the method `microversion_newer_than?`) that this only works since microversion 2.9 (and the preferred attribute only since 2.14)

**Review by Carthaca — CHANGES_REQUESTED** [review]
maybe comment or make sure technically (have a look at the method microversion_newer_than?) that this only works since microversion 2.9 (and the preferred attribute only since 2.14)

**Comment by andypf** [comment]
@Carthaca Please take a look

File: lib/fog/shared_file_system/openstack/requests/get_share_export_location.rb (hunk 1 above)
Comment (Carthaca) [reply lib/fog/shared_file_system/openstack/requests/get_share_export_location.rb]: I suggest changing to `data[:export_locations].first`

File: lib/fog/shared_file_system/openstack.rb [hunk 3]
```diff
@@ -202,6 +202,23 @@ def self.data
                   "size"        => 1
                 }
               ],
+              :export_locations      => [
+                {
+                  "path": "10.254.0.3:/shares/share-e1c2d35e-fe67-4028-ad7a-45f668732b1d",
```
Comment (Carthaca) [inline lib/fog/shared_file_system/openstack.rb]: please change to hash-rocket syntax, i.e. `=>` instead of `:`

**Review by Carthaca — CHANGES_REQUESTED** [review]

**Review by Carthaca — APPROVED** [review]

**Pull request merged**